src/main/java/hudson/plugins/sshslaves/JavaProvider.java

```diff
@@ -62,4 +71,35 @@
         return ExtensionList.lookup(JavaProvider.class);
     }
 
+    /**
+     * Gets minimal required Java version.
+     * 
+     * @return Minimal Java version required on the master and agent side.
+     *         It will be {@link #JAVA_LEVEL_7} if the core version cannot be determined due to whatever reason.
+     * @since TODO
+     * 
+     */
+    @Nonnull
+    public static VersionNumber getMinJavaLevel() {
```

Comment:
I don't get it.  The best practice is to run the same major version of java on both the master and agents, so why not just check what is running and use that.  
If you need to know the minimum then you do not need an API in core for that - you can do it today without an API by finding out the version of the classfile of `Jenkins`